ui/apps/platform/src/hooks/useEnrichNamespacesWithDeploymentCounts.ts

```diff
@@ -0,0 +1,107 @@
+import { useState, useEffect } from 'react';
+import { Namespace } from 'hooks/useFetchClusterNamespacesForPermissions';
+import { RestSearchOption } from 'services/searchOptionsToQuery';
+import { fetchDeploymentsCount } from 'services/DeploymentsService';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+export type EnrichedNamespace = {
+    metadata: {
+        id: string;
+        name: string;
+    };
+    deploymentCount;
+};
+
+export type EnrichedNamespaceResponse = {
+    loading: boolean;
+    error: string;
+    namespaces: EnrichedNamespace[];
+};
+
+export function useEnrichNamespacesWithDeploymentCounts(
+    namespaceData: Namespace[],
+    clusterId?: string
+) {
+    const [response, setResponse] = useState<EnrichedNamespaceResponse>({
+        loading: false,
+        error: '',
+        namespaces: [],
+    });
+    const [responseCount, setResponseCount] = useState<number>(0);
+    useEffect(() => {
+        if (clusterId) {
+            setResponse({
+                loading: true,
+                error: '',
+                namespaces: [],
+            });
+            const clusterSearchOptions: RestSearchOption[] = [
+                {
+                    value: 'Cluster ID:',
+                    type: 'categoryOption',
+                },
+                {
+                    value: clusterId,
+                },
+            ];
+            const enrichedNamespaces: EnrichedNamespace[] = [];
+            let shouldBreak = false;
+            let errorMessage = 'no error';
+            namespaceData.forEach((namespace) => {
```

Comment:
While I think this can be fixed by adding a call to the Deployment service that would spit out deployment counts for a list of namespaces, such API call feels a bit random. Moreover, currently namespaces are fetched for both `NetworkGraph` and `Deployment` permissions and IIUC in case the requesting principal has multiple roles, there might be `0` deployments in some namespace but we'll still render it and even show edges to it (because we can "leak" deployment name in NetworkGraph if an edge to this deployment exists). I get that this probably a very unlikely use case but it might be confusing to the user.

I guess simplest would be to just drop the counts altogether.

On the other hand, I can see how the counts can be helpful to navigate the forest of namespaces and it's sad to drop them because it's cumbersome to render them right.